inst/include/dust/distr/binomial.hpp

```diff
@@ -6,22 +6,50 @@
 namespace dust {
 namespace distr {
 
+// Faster version of pow(x, n) for integer 'n' by using
+// "exponentiation by squaring"
+// https://en.wikipedia.org/wiki/Exponentiation_by_squaring
 template <typename T>
-double binomial_inversion(rng_state_t<T>& rng_state,
-                          double n, double prob) {
-  double geom_sum = 0;
-  double num_geom = 0;
+T fast_pow(T x, int n) {
```

Comment:
I don't think so:

> If any argument has integral type, it is cast to double. If any argument is long double, then the return type Promoted is also long double, otherwise the return type is always double.

https://en.cppreference.com/w/cpp/numeric/math/pow